### ISSUE TITLE
[MNG-5884] mvn script doesn't handle directories containing spaces

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7773]
**[Stephen Kitt](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=skitt)** opened **[MNG-5884](https://issues.apache.org/jira/browse/MNG-5884?redirect=false)** and commented

find_maven_basedir() doesn't handle a parent directory working directory containing spaces. This causes a "bad variable name" warning when starting mvn.

```
me@mymachine:~/tmp/space in directory/nospace$ mvn -v
/home/me/local/maven/bin/mvn: 200: local: directory/nospace: bad variable name
Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T13:57:37+02:00)
Maven home: /home/me/local/maven
Java version: 1.7.0_79, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-openjdk-amd64/jre
Default locale: fr_FR, platform encoding: UTF-8
OS name: "linux", version: "3.13.0-62-generic", arch: "amd64", family: "unix"
```

I've submitted a patch at https://github.com/apache/maven/pull/63.

---

**Affects:** 3.3.3

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

tested successfully
merged in [f0cc5e8a](http://git-wip-us.apache.org/repos/asf/maven/commit/f0cc5e8a)
Thank you

now, I'll check on Windows to see if an equivalent fix is necessary

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

no problem found in Windows scripts
